Detectors/TOF/calibration/testWorkflow/TOFCalibCollectorWriterSpec.h

```diff
@@ -0,0 +1,127 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_CALIBRATION_TOFCALIB_COLLECTOR_WRITER_H
+#define O2_CALIBRATION_TOFCALIB_COLLECTOR_WRITER_H
+
+/// @file   TOFCalibCollectorWriterSpec.h
+/// @brief  Device to write to tree the information for TOF time slewing calibration.
+
+#include "TOFCalibration/TOFCalibCollector.h"
+#include "DataFormatsTOF/CalibInfoTOFshort.h"
+#include <TTree.h>
+#include <gsl/span>
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace calibration
+{
+
+class TOFCalibCollectorWriter : public o2::framework::Task
+{
+
+  using Geo = o2::tof::Geo;
+
+ public:
+  void createAndOpenFileAndTree()
+  {
+    TString filename = TString::Format("collTOF_%d.root", mCount);
+    LOG(DEBUG) << "opening file " << filename.Data();
+    mfileOut.reset(TFile::Open(TString::Format("%s", filename.Data()), "RECREATE"));
+    mOutputTree = std::make_unique<TTree>("treeCollectedCalibInfo", "Tree with TOF calib info for Time Slewing");
+    mOutputTree->Branch(mOutputBranchName.data(), &mPTOFCalibInfoOut);
+  }
+
+  void init(o2::framework::InitContext& ic) final
+  {
+    mCount = 0;
+    createAndOpenFileAndTree();
+    mTOFCalibInfoOut.reserve(500 * o2::tof::Geo::NCHANNELS); // should be a reasonable number of entries per channel, which allows 1 GB of data
+  }
+
+  void run(o2::framework::ProcessingContext& pc) final
+  {
+    auto collectedInfo = pc.inputs().get<std::vector<o2::dataformats::CalibInfoTOFshort>>("collectedInfo");
+    auto entriesPerChannel = pc.inputs().get<std::array<int, Geo::NCHANNELS>>("entriesCh");
```

Comment:
but you are still getting (with copy) the vector and array instead of spans, why?

Reply:
I will change it sorry, I overlooked. I even had it before, but for some reason, commented out. 
BTW, @shahor02 , you said that
"I see there are lot of commented out lines. If you are not going to revert them, better to delete."
some I need to remove (from TimeSlotCalibration), but other come from @noferini  commit. @noferini , can I uncomment the TStopwatch'es? Were they making troubles in your tests?

Chiara

Reply:
Done. With also some changes to the logs etc. 
@shahor02 , can you check that now it is better?